src/aeso_ast_infer_types.erl

```diff
@@ -253,15 +253,22 @@ bind_state(Env) ->
         false  -> Env1
     end.
 
--spec bind_field(name(), field_info(), env()) -> env().
-bind_field(X, Info, Env = #env{ fields = Fields }) ->
+-spec bind_field(append | update, name(), field_info(), env()) -> env().
+bind_field(append, X, Info, Env = #env{ fields = Fields }) ->
```

Comment:
these could be separate functions

Reply:
Done here dba72071120f798a6174e64e286df0a0e828913c